attr.go

```diff
@@ -0,0 +1,383 @@
+//Copyright 2017 GoGraphviz Authors
+//
+//Licensed under the Apache License, Version 2.0 (the "License");
+//you may not use this file except in compliance with the License.
+//You may obtain a copy of the License at
+//
+//    http)://www.apache.org/licenses/LICENSE-2.0
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS,
+//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//See the License for the specific language governing permissions and
+//limitations under the License.
+
+package gographviz
+
+import "fmt"
+
+// Attr is an attribute key
+type Attr string
+
+// NewAttr creates a new attribute key by checking whether it is a valid key
+func NewAttr(key string) (Attr, error) {
+	a, ok := validAttrs[key]
+	if !ok {
+		return Attr(""), fmt.Errorf("%s is not a valid attribute", key)
+	}
+	return a, nil
+}
+
+const (
+	DAMPING            Attr = "Damping"
+	K                  Attr = "K"
+	URL                Attr = "URL"
+	_BACKGROUND        Attr = "_background"
```

Comment:
I mean go style is technically to use CamelCase, but I actually prefer the all caps.  What do you think?